lib/matplotlib/cbook/__init__.py

```diff
@@ -2033,6 +2033,20 @@ def _setattr_cm(obj, **kwargs):
     """
     sentinel = object()
     origs = [(attr, getattr(obj, attr, sentinel)) for attr in kwargs]
+
+    # we only want to set back things that are in the instance dict or
+    # are properties.  Everything else we are either introducing to begin with
+    # or are shadowing from something later in the resolution order.
+    def filter_restores(obj, attr, val):
+        if attr in obj.__dict__:
+            return True
+
+        if isinstance(getattr(type(obj), attr), property):
```

Comment:
Perhaps safer would be:
```
if attr in type(obj).__dict__ and hasattr( type(obj).__dict__[attr], '__set__'):
```
which covers all descriptors

Reply:
I think `getattr` is better here so we see the super classes, but checking `__set__` seems better :+1:

Reply:
`getattr` will invoke some descriptors still. Perhaps `inspect.getattrstatic`?

Reply:
:facepalm: The rabbit hole gets deeper!